core/Access.php

```diff
@@ -210,27 +200,42 @@ public static function getSqlAccessSite($select)
     }
 
     /**
-     * Reload Super User access
+     * Make sure a login name is set
      *
-     * @return bool
+     * @return true
      */
-    protected function reloadAccessSuperUser()
+    protected function makeSureLoginNameIsSet()
     {
-        $this->hasSuperUserAccess = true;
-
-        try {
-            $allSitesId = Plugins\SitesManager\API::getInstance()->getAllSitesId();
-        } catch (\Exception $e) {
-            $allSitesId = array();
-        }
-        $this->idsitesByAccess['superuser'] = $allSitesId;
-
-        if(empty($this->login)) {
+        if (empty($this->login)) {
             // flag to force non empty login so Super User is not mistaken for anonymous
             $this->login = 'super user was set';
         }
+    }
 
-        return true;
+    private function loadSitesIfNeeded()
+    {
+        if ($this->hasSuperUserAccess) {
+            if (empty($this->idsitesByAccess['superuser'])) {
+                try {
+                    $allSitesId = Plugins\SitesManager\API::getInstance()->getAllSitesId();
+                } catch (\Exception $e) {
+                    $allSitesId = array();
+                }
+                $this->idsitesByAccess['superuser'] = $allSitesId;
+            }
+        } else if (isset($this->login)) {
```

Comment:
Yes it is possible and supposed to be called when `$this->login` is empty. Eg when someone is actually not logged in and also does not have anonymous access. Don't think we should throw an exception in this case.